activesupport/test/core_ext/module_test.rb

```diff
@@ -316,6 +330,14 @@ def test_delegation_with_method_arguments
     assert has_block.hello?
   end
 
+  def test_delegate_to_missing_with_method
+    assert_equal "David", DecoratedTester.new(@david).name
+  end
+
+  def test_delegate_to_missing_with_reserved_methods
+    assert_equal "David", DecoratedReserved.new(@david).name
+  end
+
```

Comment:
Missing tests for delegation to methods that don't exist and to methods with non-public visibility.